iree/compiler/Codegen/Common/ConvertToDestinationPassingStylePass.cpp

```diff
@@ -304,10 +310,84 @@ static LogicalResult duplicateInitTensorOps(OpBuilder &b,
   return success();
 }
 
+namespace {
+SmallVector<NamedAttribute> PruneAttributeList(linalg::GenericOp op) {
+  auto opAttributes = op.getAttributeNames();
+  llvm::StringSet<> elidedAttrs;
+  elidedAttrs.insert(opAttributes.begin(), opAttributes.end());
+  SmallVector<NamedAttribute> preservedAttrs;
+  for (auto attr : op->getAttrs()) {
+    if (elidedAttrs.count(attr.getName())) continue;
+    preservedAttrs.push_back(attr);
+  }
+  return preservedAttrs;
+}
+
+/// Adapts Linalg ops input operand to output operand if
+///   1. All the loops are parallel loops.
+///   2. There is only one result tensor.
+///   3. The output tensor is unused in the body computation.
+///   4. There is an input tensor that has the same indexing map and type.
+struct AdaptLinalgInputOperandToOutputOperand
+    : public OpRewritePattern<linalg::GenericOp> {
+  using OpRewritePattern<linalg::GenericOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(linalg::GenericOp op,
+                                PatternRewriter &rewriter) const override {
+    if (op.getNumLoops() != op.getNumParallelLoops()) return failure();
+    if (op->getNumResults() != 1) return failure();
+    if (!op.getRegionOutputArgs()[0].use_empty()) return failure();
+
+    OpOperand *operand = nullptr;
+    auto outputOperand = op.getOutputOperand(0);
+    SmallVector<Value> newOperands;
+    SmallVector<AffineMap> maps;
+    for (auto in : op.getInputOperands()) {
```

Comment:
It might be easier to just use `setOperand` to override the current operand used.